services/worker/src/worker/statistics_utils.py

```diff
@@ -699,3 +746,111 @@ def get_shape(example: Optional[Union[bytes, dict[str, Any]]]) -> Union[tuple[No
     @classmethod
     def transform(cls, example: Optional[Union[bytes, dict[str, Any]]]) -> Optional[int]:
         return cls.get_width(example)
+
+
+class DatetimeColumn(Column):
+    transform_column = IntColumn
+
+    @classmethod
+    def compute_transformed_data(
+        cls,
+        data: pl.DataFrame,
+        column_name: str,
+        transformed_column_name: str,
+        min_date: datetime.datetime,
+    ) -> pl.DataFrame:
+        return data.select((pl.col(column_name) - min_date).dt.total_seconds().alias(transformed_column_name))
+
+    @staticmethod
+    def shift_and_convert_to_string(base_date: datetime.datetime, seconds: Union[int, float]) -> str:
+        return datetime_to_string(base_date + datetime.timedelta(seconds=seconds))
+
+    @staticmethod
+    def get_format(data: pl.DataFrame, column_name: str) -> str:
+        values = data.filter(pl.col(column_name).is_not_null()).head(100)[column_name].to_list()
+        formats = [identify_datetime_format(value) for value in values]
+        if len(set(formats)) == 1:
+            datetime_format = formats[0]
+            if not datetime_format:
+                raise StatisticsComputationError(
+                    f"Values are datetime but format is not identified. Example: {values[0]}. "
+                )
+        else:
+            raise StatisticsComputationError(f"Multiple datetime formats detected: {set(formats)}. ")
+
+        return datetime_format
+
+    @classmethod
+    def _compute_statistics(
+        cls,
+        data: pl.DataFrame,
+        column_name: str,
+        n_samples: int,
+    ) -> DatetimeStatisticsItem:
+        nan_count, nan_proportion = nan_count_proportion(data, column_name, n_samples)
+        if nan_count == n_samples:  # all values are None
+            return DatetimeStatisticsItem(
+                nan_count=n_samples,
+                nan_proportion=1.0,
+                min=None,
+                max=None,
+                mean=None,
+                median=None,
+                std=None,
+                histogram=None,
+            )
+        original_timezone = None
+        if isinstance(data[column_name].dtype, pl.String):
+            # let polars identify format itself. provide manually in case of error
+            try:
+                original_timezone = get_timezone(data[column_name][0])
```

Comment:
I track original timezone here because polars converts datetimes to UTC when format is not provided and we need to switch it back later